search/exponential_search.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @description Exponential search algorithm for a sorted array.
+ *
+ * The algorithm searches for a specific value in a sorted array by first finding a range
```

Comment:
It would be good to explain why this may yield an advantage over just doing a binary search.

Reply:
I added a nice description.